code/game/objects/objs.dm

```diff
@@ -94,6 +94,10 @@
 	STOP_PROCESSING(SSobj, src)
 	return 0
 
+/obj/get_acid_delay()
+	. = ..()
+	if(density || isstructure(src))
+		return 4 SECONDS
```

Comment:
You probs just want the structure level of this proc to return 4 sec, rather than a structure check at obj.